helpers/integration_test.go

```diff
@@ -0,0 +1,229 @@
+		Expect(err == nil && integrationTestScenarios != nil).To(BeTrue())
+
+		for _, integrationTestScenario := range *integrationTestScenarios {
+			integrationTestScenario := integrationTestScenario //G601
+			integrationPipelineRun, err := helpers.GetLatestPipelineRunForApplicationSnapshotAndScenario(k8sClient, ctx, hasApp, hasSnapshot, &integrationTestScenario)
+			Expect(err != nil && integrationPipelineRun == nil)
+		}
+		pipelineRunOutcome, err := helpers.CalculateIntegrationPipelineRunOutcome(logger, testpipelineRun)
+		Expect(err == nil && pipelineRunOutcome).To(BeTrue())
+	})
+
+	It("should return all the required IntegrationTestScenarios used by the application", func() {
+		updatedSnapshot, err := gitops.MarkSnapshotAsPassed(k8sClient, ctx, hasSnapshot, "test passed")
+		Expect(err == nil && updatedSnapshot != nil).To(BeTrue())
+		Expect(gitops.HaveHACBSTestsFinished(hasSnapshot)).To(BeTrue())
+
+		requiredIntegrationTestScenarios, err := helpers.GetRequiredIntegrationTestScenariosForApplication(k8sClient, ctx, hasApp)
+		Expect(err == nil && requiredIntegrationTestScenarios != nil).To(BeTrue())
+		if requiredIntegrationTestScenarios != nil {
+			for _, requiredIntegrationTestScenario := range *requiredIntegrationTestScenarios {
+				requiredIntegrationTestScenario := requiredIntegrationTestScenario
+
+				integrationPipelineRuns := &tektonv1beta1.PipelineRunList{}
+				opts := []client.ListOption{
+					client.InNamespace(hasApp.Namespace),
+					client.MatchingLabels{
+						"pipelines.appstudio.openshift.io/type": "test",
+						"test.appstudio.openshift.io/snapshot":  hasSnapshot.Name,
+						"test.appstudio.openshift.io/scenario":  requiredIntegrationTestScenario.Name,
+					},
+				}
+				Eventually(func() bool {
+					err := k8sClient.List(ctx, integrationPipelineRuns, opts...)
+					return len(integrationPipelineRuns.Items) > 0 && err == nil
+				}, time.Second*10).Should(BeTrue())
+
+				integrationPipelineRun, err := helpers.GetLatestPipelineRunForApplicationSnapshotAndScenario(k8sClient, ctx, hasApp, hasSnapshot, &requiredIntegrationTestScenario)
+				Expect(err != nil && integrationPipelineRun == nil)
+
+				pipelineRunOutcome, err := helpers.CalculateIntegrationPipelineRunOutcome(logger, integrationPipelineRun)
+				Expect(err == nil && pipelineRunOutcome).To(BeTrue())
+
+				Expect(k8sClient.Delete(ctx, &integrationPipelineRuns.Items[0])).Should(Succeed())
+			}
+		}
+	})
+
+	It("ensures pipelinerun outcome when HACBSTests succeeded", func() {
+		gitops.MarkSnapshotAsPassed(k8sClient, ctx, hasSnapshot, "test passed")
+		Expect(gitops.HaveHACBSTestsSucceeded(hasSnapshot)).To(BeTrue())
+
+		pipelineRunOutcome, err := helpers.CalculateIntegrationPipelineRunOutcome(logger, testpipelineRun)
+		Expect(err == nil && pipelineRunOutcome).To(BeTrue())
+	})
+
+	It("no error from pipelinrun when HACBSTests failed", func() {
+		gitops.MarkSnapshotAsFailed(k8sClient, ctx, hasSnapshot, "test failed")
```

Comment:
Marking the snapshot as failed doesn't affect the outcome `CalculateIntegrationPipelineRunOutcome` function - it's the opposite, when the `CalculateIntegrationPipelineRunOutcome` returns `false` we then call the `MarkSnapshotAsFailed` function.
The contents of the taskRun results are what affect the outcome of `CalculateIntegrationPipelineRunOutcome`.

Reply:
Done

Reply:
Thanks!